### ISSUE TITLE
Close button on confirm dialog should have same data 

### ISSUE THREAD

**Issue by niamhmadden24** [Sage/carbon#6603]
### Description

I am using the `cancelButtonDataProps` to add a data-role to the cancel button. This should be added to the close button on the Confirm dialog so the cancel button and close button have matched data-roles 

### Reproduction

https://carbon.sage.com/?path=/docs/confirm--default-with-custom-data-tags

### Carbon version

125.12.1

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by Parsium** [comment]
Thanks for raising this @niamhmadden24 👍🏼 

An aside, if you're using React Testing Library to test the functionality of the `Confirm` close button, you can query it via its accessible name - which is set to [the locale translation key `dialog.ariaLabels.close`](https://carbon.sage.com/?path=/docs/dialog--default-story#translation-keys):

```tsx
it("renders close button when showCloseIcon prop is passed", () => {
  render(
    <Confirm
      title="Are you sure?"
      subtitle="Subtitle"
      open
      onConfirm={() => {}}
      showCloseIcon
    ></Confirm>
  );
  
  const closeButton = screen.getByRole("button", { name: "Close" }); 

  expect(closeButton).toBeVisible();
});
```

**Comment by nicktitchmarsh** [comment]
FE-6386

**Comment by Parsium** [comment]
Released part of [version 128.1.2](https://github.com/Sage/carbon/releases/tag/v128.1.2) 🚀
